client/src/Clusters/TopCharts/TopCharts.js

```diff
@@ -0,0 +1,191 @@
+import React, { Component } from "react";
+import "./SingleAppContainer.js";
+import "./TopChartsStyle.css";
+import "../styles.css";
+import SingleAppContainer from "./SingleAppContainer.js";
+ 
+class TopCharts extends Component {
+  render() {
+    const makeColumn = item => (
+      <div>
```

Comment:
Also each SingleAppContainer needs a key. You can use a unique id for this key.

```
 <SingleAppContainer
          listNum={item.listNum}
          url={item.url}
          title={item.title}
          category={item.category}
          rating={item.rating}
          key={item.id}
        />
```

Reply:
i would put the key as key = {item.title} so that the unique key would be the name of the app, but because every app (other than the first row) all have default names "App Name" . which gives me an error because they are not unique names yet.

Reply:
The title of the App is not guaranteed too be unique (i.e. search for "Messenger" on the play store). Each app already has a unique id, but since the backend/datastore is not built yet, you can add in fake ids to your mock data (i.e. "1", "2", "3", etc.)

Read more about how react uses keys here: https://reactjs.org/docs/lists-and-keys.html

Reply:
ohhh okay i see